### ISSUE TITLE
White screen (selfhosted, without docker, nginx, composer)

### ISSUE THREAD

**Issue by natrius** [firefly-iii/firefly-iii#3207]
I am running Firefly III version 5.1.1

**Description**

I was just shown the white screen. I upgraded now to version 5.1.1 just in case https://docs.firefly-iii.org/advanced-installation/upgrade#created-using-composer-create-project with this guide. I'm unfortunately still stuck with the white screen. Every other site on my server runs. 

**Extra info**
Win10 Firefox, KDE neon Firefox, KDE neon chromium, Android firefox, 

`/firefly-iii/storage/logs/ff3-cli-2020-03-25.log`

`[2020-03-25 17:48:05] local.WARNING: Database entry exists. Update to latest version (12)`

**Bonus points**

- [x] I have read the FAQ at https://bit.ly/FF3-FAQ
- [ ] I added a screenshot
- [ ] I added log files (see https://bit.ly/FF3-get-logs) (can deliver later -> update the issue-template with the correct link ? )

**Comment by JC5** [comment]
I'll update the template, thx for the suggestion.

Generally speaking a white screen is a web server issue although if other sites don't have this, that's probably not the case.

Check out https://docs.firefly-iii.org/faq/self_hosted#i-see-a-white-page-and-nothing-else as well.

**Comment by natrius** [comment]
Well, /firefly-iii/storage/logs/ff3-cli-2020-03-25.log is filled with the error mentioned above (and something before but not related to this issue). /firefly-iii/vendor/ is there. I'm using nginx and the mentioned part is there as well. 

Here is my nginx config (i use my url ofcourse instead of example.com :D ) and it worked before, btw. 

```
server {
    
    server_name firefly.example.com;
    
    root /var/www/firefly-iii/public;
    index index.php index.htm index.html;

        location / {
        try_files $uri $uri/ /index.php?$query_string;
        autoindex on;
        sendfile off;
        }

    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass unix:/run/php/php7.3-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        include fastcgi_params;
    }

    location ~ /\.ht {
        deny all;
    }

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/firefly.example.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/firefly.example.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
    
}
server {
    if ($host = firefly.example.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    server_name firefly.example.com;
    listen 80;
    return 404; # managed by Certbot
}
```

**Comment by JC5** [comment]
Is there nothing else? Because that error isn’t a problem.

**Comment by natrius** [comment]
[ff3-cli-2020-03-25.log](https://github.com/firefly-iii/firefly-iii/files/4391822/ff3-cli-2020-03-25.log)

Here is the whole file. Its the only file in /logs/ and it can be written obviously. I will enable DEBUG in the evening and restart the server again and see whats happening.

**Comment by JC5** [comment]
Are you sure you've upgraded **all** the files? This reference:

```
Command "firefly:verify" is not defined.
```

The command "firefly:verify" hasn't been part of Firefly III for quite some time now. So perhaps you're using an outdated version of the composer files?

**Comment by natrius** [comment]
`Composer version 1.9.0 2019-08-02 20:55:32` when checked. After the update `Composer version 1.10.1 2020-03-13 20:34:27`

I then updated again following all the steps here https://docs.firefly-iii.org/advanced-installation/upgrade#created-using-composer-create-project and restarted the server, again white page :(

Content of /storage/logs/ff3-cli-2020-03-27.log

```
[2020-03-27 17:10:46] local.WARNING: Database entry exists. Update to latest version (12)
[2020-03-27 17:10:56] local.WARNING: Database entry exists. Update to latest version (12)
```

**Comment by natrius** [comment]
Nginx error `sudo tail /var/log/nginx/error.log` 

```
2020/03/27 17:27:28 [error] 623#623: *150 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function bcscale() in /var/www/firefly-iii/bootstrap/app.php:37
Stack trace:
#0 /var/www/firefly-iii/public/index.php(58): require_once()
#1 {main}
  thrown in /var/www/firefly-iii/bootstrap/app.php on line 37" while reading response header from upstream, client: 77.116.180.31, server: firefly.example.com, request: "GET /index.php HTTP/2.0", upstream: "fastcgi://unix:/run/php/php7.3-fpm.sock:", host: "firefly.example.com"
```

**Comment by JC5** [comment]
- I mean the files composer.json and composer.lock. Are those the same as in the repository?
- You're missing the php-bcscale extension.

**Comment by natrius** [comment]
I installed `sudo apt install php7.3-bcmath` right now (`php7.3-bcmath` was installed already). After installing it i get errors on the page, at least :D I guess the auto-update updated on his own some packages. Now i get a red error on top `could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = installation_id and `configuration`.`deleted_at` is null limit 1)`

I installed the `php7.3-mysql` as well and now it seems to work. Kinda. Looks now like 

![image](https://user-images.githubusercontent.com/7243447/77785658-cacdab80-705c-11ea-9e83-a1a7b589a69a.png)


Fault is autoupdate updated to 7.4.
